### PR TITLE
update references in order to make this work with kafka 3.6.1

### LINKED ISSUE #1: Referenced classes were moved in recent kafka versions
I faced some issues using this library with kafka 3.6.1. It seems some classes were moved in the kafka packages. So this is not working anymore.

Is this the current implementation using azure MI tokens or is there a better way?
Is there a pre-built jar somewhere?

### PR DESCRIPTION
**Pull request by spike83**
closes #1

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -3,16 +3,16 @@ plugins {
...
 repositories {
     mavenCentral()
 }
 
 dependencies {
-    implementation group: 'org.apache.kafka', name: 'kafka-clients', version: '3.3.1'
-    implementation 'com.fasterxml.jackson.core:jackson-databind:2.13.3'
-    implementation 'org.bitbucket.b_c:jose4j:0.7.9'
+    implementation group: 'org.apache.kafka', name: 'kafka-clients', version: '3.6.1'
+    implementation 'com.fasterxml.jackson.core:jackson-databind:2.14.0-rc1'
```
Comment (javabrett) [inline build.gradle]: Any reason for needing an RC version of `jackson-databind` here - do we depend on something in the RC, or could we use a latest GA release?

File: src/main/java/io/confluent/oauth/azure/managedidentity/OAuthBearerLoginCallbackHandler.java [hunk 2]
```diff
@@ -221,8 +222,8 @@ void init(AccessTokenRetriever accessTokenRetriever, AccessTokenValidator access
     }
 
     protected AccessTokenRetriever createAccessTokenRetriever(Map<String, ?> configs,
-        String saslMechanism,
-        Map<String, Object> jaasConfig) {
+                                                              String saslMechanism,
```
Comment (javabrett) [inline src/main/java/io/confluent/oauth/azure/managedidentity/OAuthBearerLoginCallbackHandler.java]: Unexpected whitespace change.

File: src/main/java/io/confluent/oauth/azure/managedidentity/OAuthBearerLoginCallbackHandler.java [hunk 3]
```diff
@@ -43,6 +43,7 @@
 import org.apache.kafka.common.security.oauthbearer.OAuthBearerToken;
 import org.apache.kafka.common.security.oauthbearer.OAuthBearerTokenCallback;
 import org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerClientInitialResponse;
+import org.apache.kafka.common.security.oauthbearer.internals.secured.*;
 import org.apache.kafka.common.security.oauthbearer.secured.*;
```
Comment (javabrett) [inline src/main/java/io/confluent/oauth/azure/managedidentity/OAuthBearerLoginCallbackHandler.java]: This import can probably now be removed if we change to `<code>OAuthBearerValidatorCallbackHandler</code>` in the following javadocs, removing the `@code`.

**Review by javabrett — CHANGES_REQUESTED** [review]
@spike83 thanks for the PR!

File: src/main/java/io/confluent/oauth/azure/managedidentity/OAuthBearerLoginCallbackHandler.java (hunk 3 above)
Comment (spike83) [reply src/main/java/io/confluent/oauth/azure/managedidentity/OAuthBearerLoginCallbackHandler.java]: I was able to remove the `org.apache.kafka.common.security.oauthbearer.secured` but not the `org.apache.kafka.common.security.oauthbearer.internals.secured`

**Review by javabrett — APPROVED** [review]

**Comment by javabrett** [comment]
Merged - thanks again for the PR @spike83 !

**Pull request merged**